lib/buffer.js

```diff
@@ -241,6 +241,8 @@ Buffer.concat = function(list, length) {
   var pos = 0;
   for (let i = 0; i < list.length; i++) {
     var buf = list[i];
+    if (!(buf instanceof Buffer))
```

Comment:
I chose `instanceof` for consistency (`Buffer.isBuffer` is never used in this file).